llama_stack/cli/stack/build.py

```diff
@@ -74,13 +82,51 @@ def _run_stack_build_command_from_build_config(
         )
 
     def _run_stack_build_command(self, args: argparse.Namespace) -> None:
-        from llama_stack.distribution.utils.prompt_for_config import prompt_for_config
         from llama_stack.distribution.utils.dynamic import instantiate_class_type
+        from llama_stack.distribution.utils.prompt_for_config import prompt_for_config
 
         if not args.config:
-            self.parser.error(
-                "No config file specified. Please use `llama stack build /path/to/*-build.yaml`. Example config files can be found in llama_stack/distribution/example_configs"
+            name = prompt(
+                "> Please enter an unique name for identifying your Llama Stack build distribution (e.g. my-local-stack): "
             )
+            image_type = prompt(
+                "> Please enter the image type you want your distribution to be built with (docker or conda): ",
+                validator=Validator.from_callable(
+                    lambda x: x in ["docker", "conda"],
+                    error_message="Invalid image type, please enter (conda|docker)",
+                ),
+                default="conda",
+            )
+
+            cprint(
+                f"\n Llama Stack is composed of several APIs working together. Let's configure the providers (implementations) you want to use for these APIs.",
+                color="green",
+            )
+
+            providers = dict()
+            for api in Api:
+                api_provider = prompt(
+                    "> Please enter the API provider for the {} API: (default=meta-reference): ".format(
```

Comment:
shorter suggestion: `Enter API provider for {}`

we need to have a validator here absolutely

Reply:
sorry I commented prematurely :)